traffic_mirroring.go

```diff
@@ -30,36 +26,85 @@ func main() {
 	}
 
 	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
-	max_size := 10
-	packets_list := make([][]byte,max_size) 
-	current_packet := 0
+	packetsList := make(chan []byte, 22)
+	filePath := "dump_file"
+	batchSize := 20
+
+	f, err := os.Create(filePath)
+	check(err)
+	defer f.Close()
+
+	sigChan := make(chan os.Signal)
+	quit := make(chan bool)
+	signal.Notify(sigChan, os.Interrupt)
+	go handleSignal(sigChan, packetsList, quit, os.Exit)
```

Comment:
You can solve the problem of the main function returning early by executing `handleSignal` without a goroutine. In addition, call `handleSignal` after calling all the other functions, because the main code will be blocked after calling `handleSignal`.